src/controllers/PingableUserController.ts

```diff
@@ -0,0 +1,197 @@
+
+    private notifyTimeouts: {
+        [s: string]: Timeout;
+    } = {};
+
+    constructor(bot: Bot) {
+        super(bot, 'PingableUserController');
+    }
+
+    /**
+     * gets user data from the database and sets the initial last message timestamp for each user
+     */
+    public async init(): Promise<void> {
+        const db = this.bot.getDatabase();
+        const ids = CONFIG.bot.block
+            .filter((id) => id.match(/^\d+$/))
+            .map((id) => Number.parseInt(id));
+
+        await db.blockedUsers.initializeUsers(ids);
+
+        const users = await db.blockedUsers.getById(ids);
+        users.forEach((user) =>
+            user ? (this.usersLastMessage[`${user.id}`] = user.lastMessage) : undefined
+        );
+    }
+
+    /**
+     * If the sender is a user who should have pings blocked,
+     * extend the timeout for the user and notify notifiable roles
+     */
+    public async handleMessage(message: Message): Promise<boolean> {
+        if (
+            CONFIG.bot.block.indexOf(message.author.id) >= 0 &&
+            message.command === null
+        ) {
+            const now = Date.now();
+            await this.extend(message.author.id, now);
+
+            await this.notify(message);
+            this.getLogger().verbose(
+                `message sent by '${
+                    message.author.username
+                }' at ${now}... Waiting until ${
+                    now + CONFIG.bot.blockTimeout * 1000 * 60
+                }`
+            );
+
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Notifies the notifiable roles by sending a message to the indicated channels
+     * and creates a timeout that will fire once the user is no longer active
+     * @param message the message that was sent by the user that should be managed
+     * @param timeout the time, in seconds, for how long the user can be pinged
+     */
+    public async notify(message: Message, timeout = CONFIG.bot.notifyTimeout) {
+        const authorId = message.author.id;
+        let resetting = false;
+        if (this.notifyTimeouts[authorId]) {
+            resetting = true;
+            this.clearTimeout(authorId);
+        }
+
+        this.notifyTimeouts[authorId] = setTimeout(() => {
+            CONFIG.bot.notifyChannels.map(async (channelId) => {
+                const channel = message.guild?.channels.resolve(channelId) as TextChannel;
+                if (channel && channel.type === 'text') {
+                    await channel.send({
+                        content: `<@${authorId}> doesn't seem to be around anymore, you can rest your eyes`,
+                        allowedMentions: { users: [] },
+                    });
+                }
+            });
+            delete this.notifyTimeouts[authorId];
+        }, timeout * 1000 * 60);
+
+        if (resetting) return;
+
+        await Promise.all(
+            CONFIG.bot.notifyChannels.map(async (channelId) => {
```

Comment:
Please make a promise array and then `Promise.all` that array

Reply:
I'm not sure I get the point? You no likey inline mapped promises handling?

Reply:
Its nitpicking honestly, but imo its better to create an array to pass through to `Promise.all`. 

Example: 

```js
let promises = [];

for (let i = 0; i < 5; i++) {
  const promise = new Promise((resolve, reject) => {
    setTimeout(resolve, 100, 'foo');
  });
  promises.push(promise)
}

await Promise.all(promises)
```